smdebug/core/tfevent/timeline_file_writer.py

```diff
@@ -0,0 +1,172 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Writes trace events to disk in trial dir or user-specified dir."""
+
+# Standard Library
+import os
+import threading
+import time
+
+# Third Party
+import six
+
+# First Party
+from smdebug.core.locations import TraceFileLocation
+from smdebug.core.tfevent.timeline_writer import TimelineRecord, TimelineWriter
+
+
+def _get_sentinel_event():
+    """Generate a sentinel trace event for terminating worker."""
+    return TimelineRecord()
+
+
+class TimelineFileWriter:
+    """This class is adapted from EventFileWriter in Tensorflow:
+    https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/summary/writer/event_file_writer.py
+    Writes TimelineRecord to a JSON trace file.
+    The `TimelineFileWriter` class creates a timeline JSON file in the specified directory,
+    and asynchronously writes TimelineRecord to the file.
+    """
+
+    def __init__(self, path, max_queue=10, flush_secs=120):
+        """Creates a `TimelineFileWriter` and a trace event file to write to.
+        This event file will contain TimelineRecord as JSON strings, which are written to
+        disk via the write_record method.
+        The other arguments to the constructor control the asynchronous writes to
+        the event file:
+        """
+        self._path = path
+        self._event_queue = six.moves.queue.Queue(max_queue)
+        self._ev_writer = TimelineWriter(path=self._path)
+        self._ev_writer.init()
+        self._flush_secs = flush_secs
+        self._sentinel_event = _get_sentinel_event()
+        self._worker = _TimelineLoggerThread(
+            queue=self._event_queue,
+            ev_writer=self._ev_writer,
+            flush_secs=self._flush_secs,
+            sentinel_event=self._sentinel_event,
+        )
+        self._worker.start()
+
+    def write_trace_events(
+        self, training_phase="", op_name="", phase="X", timestamp=None, duration=1, args=None
+    ):
+        duration_in_us = int(duration * 100000)  # convert to micro seconds
```

Comment:
the number is missing a zero: 1.000.000 instead of 100.000

Reply:
good catch!

Reply:
fixed in the latest commit